analytic_engine/src/table_meta_set_impl.rs

```diff
@@ -294,7 +305,7 @@ impl TableMetaSet for TableMetaSetImpl {
             let version_meta = TableVersionMeta {
                 flushed_sequence,
                 files,
-                max_file_id: table_data.last_file_id(),
+                max_file_id: table_data.max_file_id().await,
```

Comment:
If we introduce a field called `max_file_id` in the `table_data`, there is no need to make this context async.